### PR TITLE
New Canvas View Pattern

### PR DESCRIPTION
**Pull request by beanh66**
## Description
Adding a new pattern for a Canvas View

### CODE REVIEW

**Comment by beanh66** [comment]
@cshinn @serenamarie125 @catrobson @LHinson @maryclarke I added the canvas view pattern, please take a look!

**Comment by maryclarke** [comment]
Couple of questions:
1. The optional Property Panel has an 'x' in the corner to close it...how do you bring it back or instantiate it for the first time
2. Can you show how the top aligned Toolbox would look non-expanded
3. What does the left aligned Toolbox look like when it has too many items to fit in one column, does it scroll vertically or expand horizontally
4. Would this pattern cover how to edit individual items or is that up to the consuming application

**Comment by beanh66** [comment]
@maryclarke Thanks for all the detailed feedback. I made a number of updates based on our conversation. Let me know what you think!

**Comment by LHinson** [comment]
@beanh66 for the automation site process, there must be a design folder so it should look like "...canvas-view/design/design.md" or "...canvas-view/design/overview.md

**Comment by maryclarke** [comment]
Looks good @beanh66

File: pattern-library/content-views/canvas-view/design.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Canvas View
+
+![Canvas Layout](img/Canvas-03.png)
+
+  1. **Editor Toolbar:**  The Editor Toolbar includes a mechanism to open and close the Toolbox as well as actions pertaining to the items on the canvas. The Toolbar can also optionally include zoom/pan actions, find, and view types.
+  2. **Canvas:** A grid based background should cover the working area of the Canvas View. Objects can be placed on the canvas to create a workflow.
```
Comment (catrobson) [inline pattern-library/content-views/canvas-view/design.md]: Workflow is a specific usage.  Can we say something more generic like what you have said on the overview page about uses for the canvas?

File: pattern-library/content-views/canvas-view/design.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+![Canvas Example with Top-aligned Toolbox](img/Canvas-06.png)
+
+1. **Toolbar:** If more than one type of toolbox must exist, a button group may be used in the Editor Toolbar to open and close one toolbox at a time.
+2. **Toolbox:** With left-aligned toolboxes, fly-out menus can be used as needed for scalability.  
+
+## Example of Top-aligned Toolbox
+![Canvas Example with Top-aligned Toolbox](img/Canvas-04.png)
+
+1. **Toolbox:** With top-aligned toolboxes, tabs and scroll capability can be used as needed for scalability.
+2. **Canvas Items:** When a user clicks on a canvas item, the selected state should be visually indicated. For cases where users will often act on a single item at one time, displaying a menu of item actions on hover is recommended.
```
Comment (catrobson) [inline pattern-library/content-views/canvas-view/design.md]: Canvas items sentence about the contextual menu reads a bit odd to me.  Maybe "For cases where there may be many actions to take on the canvas item, a contextual menu of canvas item actions on hover is recommended."

**Comment by LHinson** [comment]
@serenamarie125 if you approve, you can go ahead and hit merge! thanks @beanh66!

File: pattern-library/content-views/canvas-view/design/design.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+# Canvas View
+
+![Canvas Layout](img/Canvas-03.png)
+
+  1. **Editor Toolbar:**  The Editor Toolbar includes a mechanism to open and close the Toolbox as well as actions pertaining to the items on the canvas. The Toolbar can also optionally include zoom/pan actions, find, and view types.
+  2. **Canvas:** A grid based background should cover the working area of the Canvas View. Users can drag items from the toolbox, onto the canvas and arrange the items as needed.
+  3. **Toolbox:**
+    * All items that can be dragged onto the canvas are available in the Toolbox.
+    * The toolbox can be Left-aligned or top-aligned depending on the use case.
```
Comment (serenamarie125) [inline pattern-library/content-views/canvas-view/design/design.md]: @beanh66 can Left be all lower case ?

**Pull request merged**